### PR TITLE
Data modules

### PR DESCRIPTION
**Pull request by SaboorSohaib**
### Hi Dear :wave: 

**In this milestone, I implemented the following requirements**

- [ ] Create a new app. Give it a meaningful name.

- [ ] Initialize your project with Git.

- [ ] Make sure that your project has Postgres database set up.

- [ ] Build your project schema.

      - Create and run the necessary migration files.
      - Table and column names should match the ERD diagram.
      - note: photo for users table should be a link to an image
      - Foreign keys should be included.
      - All columns that are foreign keys should have a corresponding index.

**Kindly Check my PR.**

### CODE REVIEW

**Review by Danieldotcomcoder — COMMENTED** [review]
Hi @SaboorSohaib , 


While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

1) In your Readme file , kindly add 1-3 potential features inside your Future Features section.

2) In your Readme file, Make sure you remove the section descriptions for example "Describe how a new developer could make use of your project." should be removed and filled in with their custom content.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by julie-ify** [comment]
Hi @SaboorSohaib,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

### You have not added the future features as requested by the previous reviewer.

image

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: db/schema.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  enable_extension "plpgsql"
+
+  create_table "comments", force: :cascade do |t|
+    t.text "Text"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.bigint "AuthorId", null: false
+    t.bigint "PostId", null: false
+    t.index ["AuthorId"], name: "index_comments_on_AuthorId"
+    t.index ["PostId"], name: "index_comments_on_PostId"
```
Comment (Deepakdanger) [inline db/schema.rb]: - [ ] Kindly change `Text`, `AuthorId` & `PostId` to text, author_id, post_id.

File: db/schema.rb [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  end
+
+  create_table "likes", force: :cascade do |t|
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.bigint "PostId", null: false
+    t.bigint "AuthorId", null: false
+    t.index ["AuthorId"], name: "index_likes_on_AuthorId"
+    t.index ["PostId"], name: "index_likes_on_PostId"
```
Comment (Deepakdanger) [inline db/schema.rb]: - [ ] Kindly change `AuthorId` & `PostId` to  author_id, post_id.

File: db/schema.rb [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+  create_table "posts", force: :cascade do |t|
+    t.string "Title"
+    t.text "Text"
+    t.integer "CommentsCounter"
+    t.integer "LikesCounter"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.bigint "AuthorId", null: false
+    t.index ["AuthorId"], name: "index_posts_on_AuthorId"
```
Comment (Deepakdanger) [inline db/schema.rb]: - [ ] Kindly change `Text`, `CommentsCounter`,` LikesCounter`, `AuthorId` & `PostId` to text, comments_counter, likes_counter, author_id.

File: db/schema.rb [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  end
+
+  create_table "users", force: :cascade do |t|
+    t.string "Name"
+    t.string "Photo"
+    t.text "Bio"
+    t.integer "PostsCounter"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
```
Comment (Deepakdanger) [inline db/schema.rb]: - [ ] Kindly change `Name`, `Photo`,` Bio`, and `PostsCounter` to name, photo, Bio, posts_counter.

File: db/schema.rb [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  end
+
+  add_foreign_key "comments", "posts", column: "PostId"
+  add_foreign_key "comments", "users", column: "AuthorId"
+  add_foreign_key "likes", "posts", column: "PostId"
+  add_foreign_key "likes", "users", column: "AuthorId"
+  add_foreign_key "posts", "users", column: "AuthorId"
+end
```
Comment (Deepakdanger) [inline db/schema.rb]: - [ ] Here also kindly change same as above comments

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @SaboorSohaib , 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 
- Website looks professional :+1: 

## Required Changes ♻️
- [ ] In the rails, we always use the convention for easy access of attributes, the attributes should be small_letters & it should. Kindly change according to the below comments.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/schema.rb (hunk 5 above)
Comment (SaboorSohaib) [reply db/schema.rb]: Hi @Deepakdanger Thank you for reviewing my project. I read the changes that you requested but I followed the names that were in the diagram.
Thank you.

**Review by m05e5 — APPROVED** [review]
**Hi @SaboorSohaib 

You did great. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
CleverScornfulBuffalo-size_restricted

## Highlights

✔️ Clean and well-organized code

## Optional suggestions



_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

_**m05e5**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._**

**Pull request merged**